File: DevExpress VCL/ExpressQuantumTreeList 4/Demos/Delphi/StylesSimpleDemo/StylesSimpleDemoMain.pas
unit StylesSimpleDemoMain;

interface

uses
  Windows, Messages, SysUtils, Classes, Graphics, Controls, Forms, Dialogs,
  DB, DBTables, cxControls, ShellAPI, ComCtrls, ToolWin, ImgList, cxStyles,
  StdCtrls, ExtCtrls, Buttons,  Menus, ActnList, cxCustomData, cxGraphics,
  cxFilter, cxData, cxEdit, cxDBData, cxClasses, cxListBox, cxContainer, cxTL,
  cxMaskEdit, cxDBLookupComboBox,  cxCurrencyEdit, cxMemo, cxCheckBox,
  cxLookAndFeels, cxInplaceContainer, cxDBTL, cxTLData, cxTextEdit, cxMRUEdit,
  DemoBasicMain;

type
  TStylesSimpleDemoMainForm = class(TDemoBasicMainForm)
    cxDBTreeList: TcxDBTreeList;
    cxDBTreeList1ID: TcxDBTreeListColumn;
    cxDBTreeList1PARENTID: TcxDBTreeListColumn;
    cxDBTreeList1MANAGERID: TcxDBTreeListColumn;
    cxDBTreeList1NAME: TcxDBTreeListColumn;
    cxDBTreeList1BUDGET: TcxDBTreeListColumn;
    cxDBTreeList1LOCATION: TcxDBTreeListColumn;
    cxDBTreeList1PHONE: TcxDBTreeListColumn;
    cxDBTreeList1FAX: TcxDBTreeListColumn;
    cxDBTreeList1EMAIL: TcxDBTreeListColumn;
    cxDBTreeList1VACANCY: TcxDBTreeListColumn;
    miSeparator1: TMenuItem;
    miGridLookFeel: TMenuItem;
    miKind: TMenuItem;
    miFlat: TMenuItem;
    miStandard: TMenuItem;
    miUltraFlat: TMenuItem;
    miNativeStyle: TMenuItem;
    miSeparator2: TMenuItem;
    actShowStyleDialog: TAction;
    ShowStyleDialog1: TMenuItem;
    actFooter: TAction;
    actHeaders: TAction;
    actPreview: TAction;
    actIndicator: TAction;
    Preview1: TMenuItem;
    actIndicator1: TMenuItem;
    Header1: TMenuItem;
    Footer1: TMenuItem;
    N1: TMenuItem;
    Office111: TMenuItem;
    procedure actHeadersExecute(Sender: TObject);
    procedure actFooterExecute(Sender: TObject);
    procedure actIndicatorExecute(Sender: TObject);
    procedure actPreviewExecute(Sender: TObject);
    procedure FormShow(Sender: TObject);
    procedure actShowStyleDialogExecute(Sender: TObject);
    procedure LookAndFeelChange(Sender: TObject);
    procedure FormCreate(Sender: TObject);
    procedure cxDBTreeListInitInsertingRecord(Sender: TObject;
      AFocusedNode: TcxTreeListDataNode; var AHandled: Boolean);
  private
     procedure RestoreDefaults(Sender: TObject);
  end;

var
  StylesSimpleDemoMainForm: TStylesSimpleDemoMainForm;

implementation

uses StylesSimpleDemoData, StylesSimpleDemoEdit, StylesSimpleDemoStylesDialog;

{$R *.dfm}

procedure TStylesSimpleDemoMainForm.RestoreDefaults(Sender: TObject);
begin
  with cxDBTreeList.Styles do
  begin
    Background := nil;
    BandBackground := nil;
    BandContent := nil;
    BandHeader := nil;
    ColumnFooter := nil;
    ColumnHeader := nil;
    Content := nil;
    ContentEven := nil;
    ContentOdd := nil;
    Footer := nil;
    Inactive := nil;
    IncSearch := nil;
    Indicator := nil;
    Preview := nil;
    Selection := nil;
    StyleSheet := StylesSimpleDemoDataDM.UserStyleSheet;
  end;
end;

procedure TStylesSimpleDemoMainForm.actHeadersExecute(Sender: TObject);
begin
  cxDBTreeList.OptionsView.Headers := not cxDBTreeList.OptionsView.Headers;
  TAction(Sender).Checked := cxDBTreeList.OptionsView.Headers;
end;

procedure TStylesSimpleDemoMainForm.actFooterExecute(Sender: TObject);
begin
  cxDBTreeList.OptionsView.Footer := not cxDBTreeList.OptionsView.Footer;
  TAction(Sender).Checked := cxDBTreeList.OptionsView.Footer;
end;

procedure TStylesSimpleDemoMainForm.actIndicatorExecute(Sender: TObject);
begin
  cxDBTreeList.OptionsView.Indicator := not cxDBTreeList.OptionsView.Indicator;
  TAction(Sender).Checked := cxDBTreeList.OptionsView.Indicator;
end;

procedure TStylesSimpleDemoMainForm.actPreviewExecute(Sender: TObject);
begin
  cxDBTreeList.Preview.Visible := not cxDBTreeList.Preview.Visible;
  TAction(Sender).Checked := cxDBTreeList.Preview.Visible;
end;

procedure TStylesSimpleDemoMainForm.FormShow(Sender: TObject);
begin
{ remove/add the closing brace on this line to disable/enable the following code

  ShowMessage('WARNING: tutorial not completed. First, please apply the steps '+
              'shown in the StyleSimpleDemo.doc file');

//}
  StylesSimpleDemoStylesDialogForm.RestoreDefaults := RestoreDefaults;
  StylesSimpleDemoStylesDialogForm.Show;
end;

procedure TStylesSimpleDemoMainForm.actShowStyleDialogExecute(
  Sender: TObject);
begin
  StylesSimpleDemoStylesDialogForm.Show;
end;

procedure TStylesSimpleDemoMainForm.LookAndFeelChange(Sender: TObject);
begin
  if TMenuItem(Sender).Tag > 3 then
  begin
    cxLookAndFeelController.NativeStyle :=
      not cxLookAndFeelController.NativeStyle;
    TMenuItem(Sender).Checked := cxLookAndFeelController.NativeStyle;
  end
  else
  begin
    TMenuItem(Sender).Checked := True;
    cxLookAndFeelController.Kind := TcxLookAndFeelKind(TMenuItem(Sender).Tag);
    cxLookAndFeelController.NativeStyle := False;
    miNativeStyle.Checked := False;
  end;
end;

procedure TStylesSimpleDemoMainForm.FormCreate(Sender: TObject);
begin
  cxDBTreeList.FullExpand;
end;

procedure TStylesSimpleDemoMainForm.cxDBTreeListInitInsertingRecord(
  Sender: TObject; AFocusedNode: TcxTreeListDataNode;
  var AHandled: Boolean);
begin
  StylesSimpleDemoDataDM.SetParentValue(AFocusedNode.ParentValue);
end;

end.

File: DevExpress VCL/ExpressQuantumTreeList 4/Demos/CBuilder/StylesSimpleDemo/StylesSimpleDemoMain.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "StylesSimpleDemoMain.h"
#include "StylesSimpleDemoData.h"
#include "StylesSimpleDemoStylesDialog.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "cxClasses"
#pragma link "cxControls"
#pragma link "cxCustomData"
#pragma link "cxData"
#pragma link "cxDBData"
#pragma link "cxEdit"
#pragma link "cxFilter"
#pragma link "cxGraphics"
#pragma link "cxStyles"
#pragma link "cxLookAndFeels"
#pragma link "DemoBasicMain"
#pragma link "cxCheckBox"
#pragma link "cxCurrencyEdit"
#pragma link "cxDBLookupComboBox"
#pragma link "cxDBTL"
#pragma link "cxInplaceContainer"
#pragma link "cxMaskEdit"
#pragma link "cxTL"
#pragma link "cxTLData"
#pragma link "cxMRUEdit"
#pragma link "cxTextEdit"
#pragma resource "*.dfm"
TStylesSimpleDemoMainForm *StylesSimpleDemoMainForm;
//---------------------------------------------------------------------------
__fastcall TStylesSimpleDemoMainForm::TStylesSimpleDemoMainForm(TComponent* Owner)
  : TDemoBasicMainForm(Owner)
{
}
//---------------------------------------------------------------------------

void __fastcall TStylesSimpleDemoMainForm::FormShow(TObject *Sender)
{
  StylesSimpleDemoStylesDialogForm->RestoreDefaults = RestoreDefaults;
  StylesSimpleDemoStylesDialogForm->Show();

/* remove/add the closing slash on this line to disable/enable the following code *

  ShowMessage("WARNING: tutorial not completed. First, please apply the steps "
    "shown in the doc file");

//*/
}
//---------------------------------------------------------------------------

void __fastcall TStylesSimpleDemoMainForm::LookAndFeelChange(TObject *Sender)
{
  if (((TMenuItem*)Sender)->Tag > 3){
    cxLookAndFeelController->NativeStyle =
      !cxLookAndFeelController->NativeStyle;
    ((TMenuItem*)Sender)->Checked = cxLookAndFeelController->NativeStyle;
  }
  else{
    ((TMenuItem*)Sender)->Checked = true;
    cxLookAndFeelController->Kind = (TcxLookAndFeelKind)((TMenuItem*)Sender)->Tag;
    cxLookAndFeelController->NativeStyle = false;
    miNativeStyle->Checked = false;
  }
}
//---------------------------------------------------------------------------

void __fastcall TStylesSimpleDemoMainForm::actHeadersExecute(TObject *Sender)
{
  cxDBTreeList->OptionsView->Headers = !cxDBTreeList->OptionsView->Headers;
  ((TAction*)Sender)->Checked = cxDBTreeList->OptionsView->Headers;
}
//---------------------------------------------------------------------------

void __fastcall TStylesSimpleDemoMainForm::actFooterExecute(TObject *Sender)
{
  cxDBTreeList->OptionsView->Footer = !cxDBTreeList->OptionsView->Footer;
  ((TAction*)Sender)->Checked = cxDBTreeList->OptionsView->Footer;
}
//---------------------------------------------------------------------------

void __fastcall TStylesSimpleDemoMainForm::actIndicatorExecute(TObject *Sender)
{
  cxDBTreeList->OptionsView->Indicator = !cxDBTreeList->OptionsView->Indicator;
  ((TAction*)Sender)->Checked = cxDBTreeList->OptionsView->Indicator;
}
//---------------------------------------------------------------------------

void __fastcall TStylesSimpleDemoMainForm::actPreviewExecute(TObject *Sender)
{
  cxDBTreeList->Preview->Visible = !cxDBTreeList->Preview->Visible;
  ((TAction*)Sender)->Checked = cxDBTreeList->Preview->Visible;
}
//---------------------------------------------------------------------------

void __fastcall TStylesSimpleDemoMainForm::actShowStyleDialogExecute(TObject *Sender)
{
  StylesSimpleDemoStylesDialogForm->Show();
}
//---------------------------------------------------------------------------

void __fastcall TStylesSimpleDemoMainForm::FormCreate(TObject *Sender)
{
  cxDBTreeList->FullExpand();
}
//---------------------------------------------------------------------------

void __fastcall TStylesSimpleDemoMainForm::cxDBTreeListInitInsertingRecord(
  TObject *Sender,  TcxTreeListDataNode *AFocusedNode, bool &AHandled)
{
  StylesSimpleDemoDataDM->SetParentValue(AFocusedNode->ParentValue);
}
//---------------------------------------------------------------------------

void __fastcall TStylesSimpleDemoMainForm::RestoreDefaults(TObject *Sender)
{
  TcxTreeListStyles *AStyles = cxDBTreeList->Styles;
  AStyles->Background = NULL;
  AStyles->BandBackground = NULL;
  AStyles->BandContent = NULL;
  AStyles->BandHeader = NULL;
  AStyles->ColumnFooter = NULL;
  AStyles->ColumnHeader = NULL;
  AStyles->Content = NULL;
  AStyles->ContentEven = NULL;
  AStyles->ContentOdd = NULL;
  AStyles->Footer = NULL;
  AStyles->Inactive = NULL;
  AStyles->IncSearch = NULL;
  AStyles->Indicator = NULL;
  AStyles->Preview = NULL;
  AStyles->Selection = NULL;
  AStyles->StyleSheet = StylesSimpleDemoDataDM->UserStyleSheet;
}
//---------------------------------------------------------------------------


